src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2107,7 +2107,7 @@ BuildMapMergeJob(Query *jobQuery, List *dependentJobList, Var *partitionKey,
 static uint32
 HashPartitionCount(void)
 {
-	uint32 groupCount = ActiveReadableNonCoordinatorNodeCount();
+	uint32 groupCount = list_length(ActiveReadableNodeList());
```

Comment:
The return value of this defines the number of hash partitions. So, it doesn't matter if we use `readable` or `primary` in this context. So, it is better to keep the behaviour similar by using `ActiveReadableNodeList ()`.

(but it would probably more natural if we use an API with primary nodes in  the future)